### PR TITLE
XML API PSC Fix

### PR DESCRIPTION
**Pull request by NickGoog**
Overwrite server-returned URL with custom endpoint for complex downloads (XML API).

gsutil doesn't have unit tests for the `boto_translation.py`, and it looks like we mostly rely on `cp` integration tests to cover `GetObjectMedia`.

I'm planning a design doc for E2E tests, so we should eventually get test coverage for all this code.

### CODE REVIEW

File: gslib/boto_translation.py [hunk 1]
```diff
@@ -39,7 +39,9 @@
 
 import six
 from six.moves import http_client
+from six.moves import urllib
```
Comment (dilipped) [inline gslib/boto_translation.py:42]: Not used?

**Review by dilipped — DISMISSED** [review]

File: gslib/boto_translation.py [hunk 2]
```diff
@@ -546,6 +561,7 @@ def GetObjectMedia(self,
     if serialization_data:
       serialization_dict = json.loads(serialization_data)
       key = pickle.loads(binascii.a2b_base64(serialization_dict['url']))
+      _AddCustomEndpointToKey(key)
```
Comment (dilipped) [inline gslib/boto_translation.py:564]: Will this run for S3 interaction as well? IN that case, will we end up adding the gs_host for s3?

File: gslib/boto_translation.py (hunk 1 above)
Comment (NickGoog) [reply gslib/boto_translation.py:42]: Oops

File: gslib/boto_translation.py (hunk 2 above)
Comment (NickGoog) [reply gslib/boto_translation.py]: Yes, it looks like `BotoTranslation` is used for both S3 and GCS XML.
Added a provider check!

**Review by dilipped — APPROVED** [review]

**Pull request merged**